src/daemon/rpmostreed-deployment-utils.cxx

```diff
@@ -221,9 +242,12 @@ rpmostreed_deployment_generate_variant (OstreeSysroot    *sysroot,
 
       /* See below for base commit metadata */
       g_autoptr(GVariant) layered_metadata = g_variant_get_child_value (commit, 0);
+      g_autoptr(GVariant) detached_meta = NULL;
+      if (!ostree_repo_read_commit_detached_metadata (repo, base_checksum, &detached_meta, NULL, error))
+        return FALSE;
+
       g_variant_dict_insert (dict, "layered-commit-meta", "@a{sv}",
-                             filter ? filter_commit_meta (layered_metadata)
-                                    : layered_metadata);
+                             collect_commit_meta (layered_metadata, detached_meta, filter));
```

Comment:
Hmm, one concern with this is that it now opens the door for ambiguity if the same key exists in both commit metadata and detached metadata. I think it'd be safer and cleaner anyway to insert under separate keys instead? E.g. `(base|layered)-commit-detached-meta`.

Reply:
I can do that but it's a potential Principle Of Least Astonishment violation for people downstream of this. I can see the arguments either way, I just went with "break people the least" for now.

Another option would be to tweak the `collect_commit_meta` function so that it'll warn on name collisions?

Reply:
I think future-proofing-wise it'd be cleaner to keep them separate from the start. The status quo is that `(base|layered)-commit-meta` is a direct map of commit metadata so we're retaining the same semantics and this is purely adding a new thing.